resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -73,6 +76,30 @@ private[spark] class Client(
   private val driverJavaOptions = sparkConf.get(
     org.apache.spark.internal.config.DRIVER_JAVA_OPTIONS)
 
+  // create resource of kind - SparkJob representing the deployed spark app
+  private val statusMap = Map(
+    STATUS_CREATION_TIMESTAMP -> Calendar.getInstance().getTime().toString(),
+    STATUS_COMPLETION_TIMESTAMP -> STATUS_NOT_AVAILABLE,
+    STATUS_DRIVER -> kubernetesDriverPodName,
+    STATUS_DRIVER_IMAGE -> driverDockerImage,
+    STATUS_EXECUTOR_IMAGE -> sparkConf.get(EXECUTOR_DOCKER_IMAGE),
+    STATUS_JOB_STATE -> JobState.QUEUED,
+    STATUS_DESIRED_EXECUTORS -> sparkConf.getInt("spark.executor.instances", 1),
+    STATUS_CURRENT_EXECUTORS -> 0,
+    STATUS_DRIVER_UI -> STATUS_PENDING
+  )
+
+  // Failure might be due to TPR inexistence or maybe we're stuck in the 10 minute lag
```

Comment:
That's my bad. It's 10s. Here's a comment on an [issue ](https://github.com/kubernetes/features/issues/95#issuecomment-255575716) opened on TPRs